### DISCUSSION TITLE
Experimental noScripts

### DISCUSSION THREAD

**Discussion by green-cats** [nuxt/nuxt#16743] [Questions] [answered]
Can I enable scripts for specific components? That would be great, as it up performance so much.

**Comment by TheAlexLichter** [comment, marked as answer]
Follow https://github.com/nuxt/framework/pull/5688 for more ☺️

**Reply by green-cats** [reply]
@manniL thank you mate!

**Comment by marvincaspar** [comment]
I'm looking for this feature too. But in my case, I have a fully static page and just need a click event listener for toggling the mobile navigation. I have no server component.

**Reply by TheAlexLichter** [reply]
In your case, you can use the feature introduced in https://github.com/nuxt/framework/pull/7248 together with a nitro hook to add the event listener script eventually 😋

**Reply by marvincaspar** [reply]
i didn't get it. How should I add the event listener to nitro? 

Here is my js code for the click event:

```js
<script setup lang="ts">
  defineProps({ blok: Object });

  const primaryHeader = ref<HTMLElement | null>(null);
  const navToggle = ref<HTMLElement | null>(null);
  const primaryNav = ref<HTMLElement | null>(null);
  onMounted(() => {
    navToggle.value?.addEventListener("click", () => {
      primaryNav.value?.hasAttribute("data-visible")
        ? navToggle.value?.setAttribute("aria-expanded", "false")
        : navToggle.value?.setAttribute("aria-expanded", "true");
      primaryNav.value?.toggleAttribute("data-visible");
      primaryHeader.value?.toggleAttribute("data-overlay");
    });
  });
</script>
```

**Reply by TheAlexLichter** [reply]
You would not do it in Vue. Instead, you would need to create a Nitro plugin that hooks into [`render:html`](https://v3.nuxtjs.org/api/advanced/hooks/#nuxt-hooks-build-time) and adds vanilla JavaScript.

**Reply by TheAlexLichter** [reply]
Example: https://stackblitz.com/edit/https-github-comnuxtframeworkdiscussions7747-3740973?file=app.vue,server%2Fplugins%2Frender.ts,public%2Fmain.js,nuxt.config.ts,package.json,README.md

**Reply by marvincaspar** [reply]
Thanks, i tried it and it works. Thanks

**Reply by larxn** [reply]
As a complement to @manniL's [demo](https://stackblitz.com/edit/https-github-comnuxtframeworkdiscussions7747-3740973?file=app.vue,server%2Fplugins%2Frender.ts,public%2Fmain.js,nuxt.config.ts,package.json,README.md), you can also specify which pages will receive the script injection:

```ts
export default defineNitroPlugin((nitroApp) => {
  nitroApp.hooks.hook('render:html', (html, ctx) => {
    if (ctx.event.path === '/') {
      html.head.push(
        '<script src="/js/analytics.js"></script>',
      )
    }
  })
})
```
In this case, the script will only be injected into my home page (`/`).
